pkgs/applications/window-managers/wayfire/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv, fetchFromGitHub
+, meson, pkgconfig, ninja
+, wayland, wayland-protocols
+, cairo, glm
+, libevdev, freetype, libinput
+, pixman, libxkbcommon, libdrm
+, libjpeg, libpng
+, libGL, mesa
+, libcap, xcbutilerrors, xcbutilwm, libxml2
+, buildDocs ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "wayfire";
+  version = "8aa8260972edaba6d237e1a61217a4d2303edc3e";
+
+  src = fetchFromGitHub {
+    owner = "WayfireWM";
+    repo = "wayfire";
+    rev = version;
+    sha256 = "sha256-r9k8e0LCDxr8tEZvcBbkvZ2RqcKXYX+/fxalKoqT15U=";
+    fetchSubmodules = true;
```

Comment:
Can we package the submodules as different packages?

Reply:
Yes we can in fact I was doing that before.

Reply:
There is a problem that wf-utils and wf-touch are expected to a subproject. The meson.build fails if we pass it as a dependency. We can do something like [this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome-3/core/gvc-with-ucm-prePatch.nix) to avoid pulling down submodules automatically.

Reply:
I am not sure if this is the ideal solution. Do you know if upstream would support using system libraries? Otherwise I think this is fine if you do not know a better solution.

Reply:
Actually the I mentioned didn't work the build is failing. Also the submodules wayfire requires provide utility classes and support for touch. We could also try patching meson.build to add the wf-touch and wf-utils as dependencies and not as a submodule but I am not familiar with meson.build (I am just firing blind arrows after reading the meson.build file).

Reply:
Ok after looking into the issue why the build was failing was due to wlroots in nixpkgs is incompatible with the version wayfire is using. So we will have to use the fetchSubmodules.

Reply:
Normally we try not to vendor packages that are already included in nixpkgs to reduce closure size.